core/2d/CCParticleSystem.cpp

```diff
@@ -2357,14 +2357,17 @@ void ParticleEmissionMaskCache::bakeEmissionMask(std::string_view maskName,
     desc.points = std::move(points);
 
     iter->second = desc;
+
+    CCLOG("Particle emission mask '%s' baked (%dx%d), %d samples generated taking %.2fmb of memory.",
+          std::string(maskName).c_str(), w, h, desc.points.size(), desc.points.size() * 8 / 1e+6);
 }
 
 const ParticleEmissionMaskDescriptor& ParticleEmissionMaskCache::getEmissionMask(std::string_view maskName)
 {
     auto iter = this->masks.find(maskName);
     if (iter == this->masks.end())
     {
-        ParticleEmissionMaskDescriptor desc;
+        ParticleEmissionMaskDescriptor desc{};
```

Comment:
Not emplace to masks?

Reply:
We don't need to emplace here because we are getting data from masks map not modifying it.

Reply:
I didn't actually think about it, it should be stored in the masks even though it's a dummy mask as to no overflow memory.